.github/workflows/docs-build-push.yml

```diff
@@ -50,7 +50,26 @@ env:
   DOMAIN_PROD: "docs.nginx.com"
 
 jobs:
+  checks:
+    name: Checks and variables
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+    outputs:
+      forked_workflow: ${{ steps.vars.outputs.forked_workflow }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
Is there a reason why we use a specific version of checkout?

Reply:
Yes, this is how dependabot locks down versions. WIth the dependabot action runs, it'll create a PR for version updates (https://github.com/nginxinc/docs-actions/blob/main/.github/dependabot.yml). 
On a version bump, it'll update to the specific commit hash, and add the release version as a comment.
In theory, this should mean if someone was to exploit release `4.1.7` we'd see the commit hash change, but no the version, and know not to accept it. Similar to a package lock for npm. Dependabot can also automatically update package.json's btw, which is useful!